pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -48,22 +48,14 @@ stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
-  ## Patch
-
   patches = [
     ./docbook2texi.patch
     ./git-sh-i18n.patch
-    ./ssh-path.patch
     ./git-send-email-honor-PATH.patch
     ./installCheck-path.patch
   ];
 
   postPatch = ''
```

Comment:
This breaks cloning ssh repos. I don't think we should do this. I think we should rethink the possible solutions and go though an RFC before breaking tons of problems to work around a problem in Nix.

Reply:
@kevincox what does this break? Git works and if you have SSH installed cloning via SSH works too. IMO that's how it should be. If there are enough relevant use-cases we could also add a wrapped Git to `all-packages.nix`.

Reply:
The key phrase is "if you have SSH installed". If I am using `gitFull` I expect to be able to use all of the features of Git. Not get errors about missing dependencies.

I think a wrapped package for `gitFull` would be acceptable. That way the rebuild is trivial without having a disfunctional git package.

Reply:
> The key phrase is "if you have SSH installed".

Yes, and my point is that this should be the intended behaviour (SSH is a completely optional run time dependency and there are multiple SSH client packages available (plus different versions and configurations)).

> I think a wrapped package for gitFull would be acceptable.

That sounds fair to me.

Reply:
I made the patch conditional for now because I don't want to deal with the implications of wrapping the git binary.

Reply:
Thanks. I think this is a good tradeoff.